src/systems/drive_to_pose_controller/DriveToPoseController.cc

```diff
@@ -0,0 +1,252 @@
+  public: void OnCurrentPose(const msgs::Pose_V &_msg);
+
+  /// \brief Calculate velocity and publish a twist message
+  public: void CalculateVelocity();
+
+  /// \brief Normalize the angles between -pi to pi
+  /// \param[in] _angle Angle to normalize
+  public: void NormalizeAngle(double &_angle);
+
+  /// \brief Gazebo communication node
+  public: transport::Node node;
+
+  /// \brief Model interface
+  public: Model model{kNullEntity};
+
+  /// \brief Velocity message publisher.
+  public: transport::Node::Publisher velocityPublisher;
+
+  /// \brief Pose reached publisher
+  public: transport::Node::Publisher poseReachedPublisher;
+
+  /// \brief Current pose of the model
+  public: std::optional<math::Pose3d> currentPose;
+
+  /// \brief Target pose of the model
+  public: std::optional<math::Pose3d> targetPose;
+
+  /// \brief Proportional gain for angular velocity
+  public: double angularPGain{2.0};
+
+  /// \brief Proportional gain for linear velocity
+  public: double linearPGain{1.0};
+
+  /// \brief Allowable angular deviation
+  public: double angularDeviation{0.05};
+
+  /// \brief Allowable linear deviation
+  public: double linearDeviation{0.1};
+
+  /// \brief Mutex to protect current and target poses
+  public: std::mutex mutex;
+};
+
+//////////////////////////////////////////////////
+DriveToPoseController::DriveToPoseController()
+    : dataPtr(std::make_unique<DriveToPoseControllerPrivate>())
+{
+  // Do nothing
+}
+
+//////////////////////////////////////////////////
+void DriveToPoseController::Configure(
+    const Entity& _entity,
+    const std::shared_ptr<const sdf::Element>& _sdf,
+    EntityComponentManager& _ecm,
+    EventManager& /*_eventMgr*/)
+{
+  this->dataPtr->model = Model(_entity);
+  if (!this->dataPtr->model.Valid(_ecm))
+  {
+    gzerr << "This plugin should be attached to a model entity. "
+          << "Initialization failed." << std::endl;
+    return;
+  }
+
+  // Get linear P gain if available
+  if (_sdf->HasElement("linear_p_gain"))
+  {
+    this->dataPtr->linearPGain = _sdf->Get<double>("linear_p_gain");
+  }
+
+  // Get angular P gain if available
+  if (_sdf->HasElement("angular_p_gain"))
+  {
+    this->dataPtr->angularPGain = _sdf->Get<double>("angular_p_gain");
+  }
+
+  // Get allowable linear deviation if available
+  if (_sdf->HasElement("linear_deviation"))
+  {
+    this->dataPtr->linearDeviation = _sdf->Get<double>("linear_deviation");
+  }
+
+  // Get allowable angular deviation if available
+  if (_sdf->HasElement("angular_deviation"))
+  {
+    this->dataPtr->angularDeviation = _sdf->Get<double>("angular_deviation");
+  }
+
+  // Topic namespace for publish and subscribe
+  std::string topicNamespace = "/model/" + this->dataPtr->model.Name(_ecm);
+
+  // Subscribe to pose topics
+  this->dataPtr->node.Subscribe(
+    topicNamespace + "/cmd_pose", &DriveToPoseControllerPrivate::OnCmdPose,
+    this->dataPtr.get());
+
+  this->dataPtr->node.Subscribe(
```

Comment:
Same thing for subscribers, we can detect if there are publishers. If no publishers inform the user that we need to load the correct set of plugins. (We could also automatically load them if that is something you are willing to look into).